src/coreclr/vm/cgensys.h

```diff
@@ -97,8 +97,14 @@ inline void GetSpecificCpuInfo(CORINFO_CPU * cpuInfo)
 // MSVC directly defines intrinsics for __cpuid and __cpuidex matching the below signatures
 // We define matching signatures for use on Unix platforms.
 
-extern "C" void __stdcall __cpuid(int cpuInfo[4], int function_id);
-extern "C" void __stdcall __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
+#ifdef TARGET_X86
+#define CALLCONV __cdecl
+#else
+#define CALLCONV __stdcall
+#endif
+
+extern "C" void CALLCONV __cpuid(int cpuInfo[4], int function_id);
+extern "C" void CALLCONV __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
```

Comment:
```suggestion
extern "C" void __cpuid(int cpuInfo[4], int function_id);
extern "C" void __cpuidex(int cpuInfo[4], int function_id, int subFunction_id);
```
Does this work?

It is best to avoid explicit cdecl vs. stdcall calling conventions on non-Windows.

Reply:
It compiles on linux x86 and osx arm64. Lets see what CI thinks. :)